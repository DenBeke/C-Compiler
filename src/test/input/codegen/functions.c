// Functions

#include <stdio.h>

// Fibonacci
int fib(int n) {
    
    if(n == 0) {
        return 0;
    }
    else if(n == 1) {
        return 1;
    }
    
    return fib(n - 1) + fib(n - 2);
    
}

// Factorial
int fac(int n) {
    if(n == 1 || n == 0) {
        return 1;
    }
    return n * fac(n - 1);
}


void main() {
    
    // Nested function
    void hello_world() {
        print("Hello World!\n");
    }
    
    hello_world();
    
    printf("Fib = %d\n", fib(10));
    
    printf("Fac = %d\n", fac(4));
    
    printf(":)  = %d\n", fib(fac(3)));
    
    int f = fac(3);
    printf(":)  = %d\n", fib(f));
    
}