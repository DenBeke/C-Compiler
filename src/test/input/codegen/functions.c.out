Hello World!
Fib = 55
Fac = 24
:)  = 8
:)  = 8
